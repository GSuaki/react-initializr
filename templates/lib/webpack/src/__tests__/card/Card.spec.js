require('jsdom-global/register');
const React = require('react');
const { mount } = require('enzyme');
const renderer = require('react-test-renderer');

const Card = require('../../card/src/Card').default;

const onCardClick = jest.fn()

const mockProps = {
  showPlaceholder: false,
  onCardClick
};

const mountComponent = (props = mockProps) => {
  const component = (
    <Card { ...props } >
      <div className="content">Hello</div>
    </Card>
  )

  const element = mount(component)

  return { component, element }
}

describe('Components -> Card', () => {
  it('render', () => {
    const { component, element } = mountComponent();

    expect(element.find('.card')).toHaveLength(1);
    expect(element.find('.content')).toHaveLength(1);

    expect(renderer.create(component).toJSON()).toMatchSnapshot();
  });
  
  it('render with placeholder', () => {
    const { component, element } = mountComponent({
      ...mockProps,
      showPlaceholder: true
    })

    expect(element.find('.card')).toHaveLength(1);
    expect(element.find('.content')).toHaveLength(0);
    expect(element.find('.awesome-placeholder')).toHaveLength(1);

    expect(renderer.create(component).toJSON()).toMatchSnapshot();
  });
  
  it('simulate card click event: onCardClick', () => {
    const { element } = mountComponent();

    element.find('.card').simulate('click');
    expect(onCardClick).toHaveBeenCalledTimes(1);
  });
});
