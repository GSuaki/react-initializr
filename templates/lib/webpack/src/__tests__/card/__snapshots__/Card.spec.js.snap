// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components -> Card render 1`] = `
<div
  className="card"
  onClick={[MockFunction]}
>
  <div
    className="content"
  >
    Hello
  </div>
</div>
`;

exports[`Components -> Card render with placeholder 1`] = `
<div
  className="card"
  onClick={[MockFunction]}
>
  <div
    className="awesome-placeholder show-loading-animation"
  >
    <div
      className="rect-shape"
      style={
        Object {
          "backgroundColor": "#e4e4e4",
          "height": 158,
          "marginRight": 10,
          "width": "100%",
        }
      }
    />
    <div
      className="text-block"
      style={
        Object {
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <div
        className="text-row"
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "height": "1em",
            "marginTop": 0,
            "maxHeight": "7.6923076923076925%",
            "width": "97%",
          }
        }
      />
      <div
        className="text-row"
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "height": "1em",
            "marginTop": "0.7em",
            "maxHeight": "7.6923076923076925%",
            "width": "100%",
          }
        }
      />
      <div
        className="text-row"
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "height": "1em",
            "marginTop": "0.7em",
            "maxHeight": "7.6923076923076925%",
            "width": "94%",
          }
        }
      />
      <div
        className="text-row"
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "height": "1em",
            "marginTop": "0.7em",
            "maxHeight": "7.6923076923076925%",
            "width": "90%",
          }
        }
      />
      <div
        className="text-row"
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "height": "1em",
            "marginTop": "0.7em",
            "maxHeight": "7.6923076923076925%",
            "width": "98%",
          }
        }
      />
      <div
        className="text-row"
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "height": "1em",
            "marginTop": "0.7em",
            "maxHeight": "7.6923076923076925%",
            "width": "95%",
          }
        }
      />
      <div
        className="text-row"
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "height": "1em",
            "marginTop": "0.7em",
            "maxHeight": "7.6923076923076925%",
            "width": "98%",
          }
        }
      />
    </div>
  </div>
</div>
`;
